server/src/pages/shot/editor.js

```diff
@@ -698,12 +700,46 @@ class ColorPicker extends React.Component {
     </div>
   }
 
+  componentDidUpdate() {
+    if (this.state.pickerActive) {
+      document.addEventListener("mousedown", this.clickMaybeClose);
+      document.addEventListener("keyup", this.keyMaybeClose);
+    } else {
+      document.removeEventListener("mousedown", this.clickMaybeClose);
+      document.removeEventListener("keyup", this.keyMaybeClose);
+    }
```

Comment:
As written, the listeners could be added and removed unnecessarily when the component is updated for a change unrelated to the picker.  `componentDidUpdate()` receives the old props and old state as its arguments, so you can check if the `pickerActive` state has changed and then add/remove accordingly.

Reply:
Oops, sorry, I didn't realize this is in the picker component itself.  I think that'd be okay then.